libcst/metadata/tests/test_scope_provider.py

```diff
@@ -1310,3 +1310,51 @@ def test_gen_dotted_names(self) -> None:
             )
         }
         self.assertEqual(names, {"a.b.c", "a.b", "a"})
+
+    def test_ordering(self) -> None:
```

Comment:
Can you add another test case that the Assignment is defined after the access in a different scope.
E.g.
```
def f():
    print(ab)  # this access should refer to the below assign

ab = 1
```

```
def f(ab):
    print(ab)  # this access should refer to the above assign

ab = 1
```

Reply:
Yep, will do